### ISSUE TITLE
Understanding testStateMachine

### ISSUE THREAD

**Issue by hally9k** [MicheleBertoli/react-automata#72]
Firstly - Awesome work here man!!  ❤️❤️❤️

Secondly - I am trying to understand  `testStateMachine` from reading the README and I'll admit I'm slightly stumped. So I thought I'd reach out for some help and then maybe I can return the favour by fleshing out the README to be more friendly to slow people, like myself.

The main thing I'm having trouble with is how to pass props into the component. After reading the README I thought that the `initialData` object in the fixtures would be passed into the component as props, but this doesn't seem to work. Am I missing something?

**Comment by MicheleBertoli** [comment]
Thank you very much for opening this issue, and for the kind words, @hally9k.

Unless I'm missing something, `initialData` is what you are looking for.

For example:

```js
test('props work', () => {
  const myStatechart = {
    initial: 'myState',
    states: { myState: {} },
  }
  const MyComponent = props => <div>{props.myProp}</div>
  const StateMachine = withStateMachine(myStatechart)(MyComponent)
  const initialData = { myProp: 'Hello, Props!' }

  testStateMachine(StateMachine, { fixtures: { initialData } })
})
```

Generates the following snapshot:

```js
exports[`props work: myState 1`] = `
<div>
  Hello, Props!
</div>
`;
```

Please let me know if this answers your question, and any contribution to the README is more than welcome.

**Comment by hally9k** [comment]
It appears to be only when I pass nested objects as props. This is an example that fails:

```js
it('props work', () => {
        const myStatechart = {
            initial: 'myState',
            states: { myState: {} }
        }

        const MyComponent = (props) => <div>{props.node.content.text}</div>

        const StateMachine = withStateMachine(myStatechart)(MyComponent)
        const initialData = { node: { content: { text: 'woot' } } }

        testStateMachine(StateMachine, { fixtures: { initialData } })
    })
```

**Comment by MicheleBertoli** [comment]
Aha, good catch - Fixed in `v4.0.4`.
Thank you very much!
